### PR TITLE
feat(sdk): expose metagen to `typegraph/sdk`

### PR DESCRIPTION
**Pull request by afmika**
Expose metagen features to `typegraph/sdk`

Depends on #707 and #696

#### Migration notes

None

## Summary by CodeRabbit

- **New Features**
  - Introduced new `Metagen` class for code generation tasks in both TypeScript and Python SDKs.
  - Added functionality for defining policies and structures for deployment examples using Node.js.

- **Improvements**
  - Enhanced `Metagen` class with methods for simulating and executing code generation tasks.
  - Simplified file reading and writing functions for better performance and maintainability.

- **Bug Fixes**
  - Refined `compress_and_encode` function to streamline file handling processes.

- **Tests**
  - Added comprehensive tests for `Metagen` functionality in both TypeScript and Python SDKs.

### CODE REVIEW

File: libs/metagen/src/tests/mod.rs [hunk 1]
```diff
@@ -38,6 +38,7 @@ pub struct E2eTestCase {
     pub build_fn: fn(BuildArgs) -> BoxFuture<anyhow::Result<()>>,
 }
 
+#[cfg(feature = "multithreaded")]
```
[the remark this answers is not in the document]
Comment (afmika) [reply libs/metagen/src/tests/mod.rs:41]: Actually, it is defined at `libs/metagen/Cargo.toml`

File: typegate/tests/metagen/metagen_test.ts [hunk 2]
```diff
@@ -59,3 +59,63 @@ members = ["mdk/"]
...
+  );
+
+  assertEquals(
+    (await Meta.cli({}, ...`-C ${tmpDir} gen mdk my_target`.split(" "))).code,
+    0,
+  );
+});
+
+Meta.test("Metagen within sdk", async (t) => {
+  await t.should("Run metagen within typescript", async () => {
```
Comment (Yohe-Am) [inline typegate/tests/metagen/metagen_test.ts:98]: question: test for the python sdk?

**Review by Yohe-Am — APPROVED** [review]

File: typegate/tests/metagen/metagen_test.ts (hunk 2 above)
Comment (afmika) [reply typegate/tests/metagen/metagen_test.ts:98]: I am awaiting for #716

Comment (afmika) [reply typegate/tests/metagen/metagen_test.ts:98]: Or nvm, I will just spawn a python process and use the stdout.

**Review by zifeo — APPROVED** [review]
👏

**Pull request merged**